ZeekPluginDynamic.cmake

```diff
@@ -1,362 +1,145 @@
-
-   add_definitions(-DZEEK_PLUGIN_INTERNAL_BUILD=false)
-
-   add_custom_target(generate_outputs)
-
-   if ( BRO_PLUGIN_ENABLE_DEBUG )
-       set(ENABLE_DEBUG true)
-       set(CMAKE_BUILD_TYPE Debug)
-   endif ()
-
-   include(SetDefaultCompileFlags)
-
-else ()
-    set(BRO_PLUGIN_BASE        "${CMAKE_CURRENT_BINARY_DIR}"         CACHE INTERNAL "" FORCE)
-    set(BRO_PLUGIN_LIB         "${CMAKE_CURRENT_BINARY_DIR}/lib"     CACHE INTERNAL "" FORCE)
-    set(BRO_PLUGIN_BIF         "${BRO_PLUGIN_LIB}/bif"               CACHE INTERNAL "" FORCE)
-    set(BRO_PLUGIN_MAGIC       "${BRO_PLUGIN_BASE}/__bro_plugin__"   CACHE INTERNAL "" FORCE)
-    set(BRO_PLUGIN_README      "${BRO_PLUGIN_BASE}/README"           CACHE INTERNAL "" FORCE)
-    set(BRO_PLUGIN_SCRIPTS     "${BRO_PLUGIN_BASE}/scripts"          CACHE INTERNAL "" FORCE)
-    set(BRO_PLUGIN_SCRIPTS_SRC "${CMAKE_CURRENT_SOURCE_DIR}/scripts" CACHE INTERNAL "" FORCE)
-endif ()
-
+include(FindClangTidy)
 include(GetArchitecture)
 
-function(bro_plugin_bif_dynamic)
-    foreach ( bif ${ARGV} )
-        bif_target(${bif} "plugin" ${_plugin_name} ${_plugin_name_canon} FALSE)
-        list(APPEND _plugin_objs ${BIF_OUTPUT_CC})
-        list(APPEND _plugin_deps ${BIF_BUILD_TARGET})
-        set(_plugin_objs "${_plugin_objs}" PARENT_SCOPE)
-        set(_plugin_deps "${_plugin_deps}" PARENT_SCOPE)
-    endforeach ()
-endfunction()
-
-function(bro_plugin_link_library_dynamic)
-    foreach ( lib ${ARGV} )
-        set(_plugin_libs ${_plugin_libs} ${lib} CACHE INTERNAL "dynamic plugin libraries")
-    endforeach ()
-endfunction()
-
-function(bro_plugin_end_dynamic)
-    # Create the dynamic library/bundle.
-    add_library(${_plugin_lib} MODULE ${_plugin_objs})
-    set_target_properties(${_plugin_lib} PROPERTIES LIBRARY_OUTPUT_DIRECTORY "${BRO_PLUGIN_LIB}")
-    set_target_properties(${_plugin_lib} PROPERTIES PREFIX "")
-    # set_target_properties(${_plugin_lib} PROPERTIES ENABLE_EXPORTS TRUE)
-
-    add_dependencies(${_plugin_lib} generate_outputs)
+# Sets `target` to contain the CMake target name for a dynamic plugin.
+macro(zeek_get_dynamic_plugin_target target ns name)
+    set(${target} "${ns}_${name}")
+endmacro()
 
-    if ( _plugin_deps )
-        add_dependencies(${_plugin_lib} ${_plugin_deps})
-    endif()
+# Implements the dynamically linked version of zeek_add_plugin.
+function(zeek_add_dynamic_plugin ns name)
+    # Sanity check: need ZEEK_PLUGIN_SCRIPTS_PATH.
+    if (NOT EXISTS "${ZEEK_PLUGIN_SCRIPTS_PATH}")
+        message(FATAL_ERROR "Cannot build dynamic plugins: ZEEK_PLUGIN_SCRIPTS_PATH is undefined or invalid")
+    endif ()
 
-    target_link_libraries(${_plugin_lib} ${_plugin_libs})
-    if ( MSVC )
-        target_link_libraries(${_plugin_lib} ${ZEEK_LIBRARY} ws2_32)
+    # Helper variables.
+    zeek_get_dynamic_plugin_target(target_name ${ns} ${name})
+    set(full_name   "${ns}::${name}")
+    set(canon_name  "${ns}_${name}")
```

Comment:
This is equivalent to `${target_name}`

Reply:
True, but the canonical name has meaning beyond CMake (used as file name for the tarball). It's currently the same as the target name, but I think being explicit here still makes sense.

Reply:
Works for me.